libraries/rush-lib/src/logic/buildCache/ProjectBuildCache.ts

```diff
@@ -13,33 +13,37 @@ import type { ICloudBuildCacheProvider } from './ICloudBuildCacheProvider';
 import type { FileSystemBuildCacheProvider } from './FileSystemBuildCacheProvider';
 import { TarExecutable } from '../../utilities/TarExecutable';
 import { EnvironmentVariableNames } from '../../api/EnvironmentConfiguration';
+import type { Operation } from '../operations/Operation';
 
-export interface IProjectBuildCacheOptions {
+export interface IOperationBuildCacheOptions {
   /**
    * The repo-wide configuration for the build cache.
    */
   buildCacheConfiguration: BuildCacheConfiguration;
-  /**
-   * The project to be cached.
-   */
-  project: RushConfigurationProject;
   /**
    * Value from rush-project.json
    */
   projectOutputFolderNames: ReadonlyArray<string>;
-  /**
-   * The hash of all relevant inputs and configuration that uniquely identifies this execution.
-   */
-  operationStateHash: string;
   /**
    * The terminal to use for logging.
    */
   terminal: ITerminal;
+}
+
+export type IProjectBuildCacheOptions = IOperationBuildCacheOptions & {
+  /**
+   * The project to be cached.
+   */
+  project: RushConfigurationProject;
```

Comment:
We can solve the need for this and `phaseName` by finally making `associatedProject` and `associatedPhase` required again on `Operation`.